src/controllers/BaseController.ts

```diff
@@ -18,11 +18,14 @@ import { ValidationResult } from 'app/utils/validation/ValidationResult';
 
 export type FormSanitizeFunction<T> = (body: T) => T;
 
-const createChangeModeAwareNavigationProxy = (step: Navigation): Navigation => {
+const createChangeModeAwareNavigationProxy = (step: Navigation, changeModeAction: string | undefined): Navigation => {
     return new Proxy(step, {
-        get(target: Navigation, propertyName: 'previous' | 'next'): any {
+        get(target: Navigation, propertyName: keyof Navigation): any {
```

Comment:
I feel like you could do a lot more with this. Since you pass the propertyName, you can can also have the controller define what happens if previous and next are called when `cm === 1`

Reply:
ATM you're doing that manually in the view files

Reply:
Good point, this would be a lot cleaner. I'll see if i can get this to work as some of our controllers have more than one action and some templates link to another controller.

Reply:
I've spent quite a bit of time trying to get this to work, tried the following:
* Removed the manual check of change mode from view files
* Defined previous and next URIs when in change mode in EvidenceUploadController and EvidenceRemovalController

Unfortunately without the `cm=1` parameter in the request, the controllers were not able to identify whether we were in change mode (new previous and next were not read). Maybe i'm missing something?